pytorch_lightning/trainer/training_loop.py

```diff
@@ -556,7 +556,8 @@ def check_checkpoint_callback(self, should_check_val):
             [c.on_validation_end(self, self.get_model()) for c in checkpoint_callbacks]
 
     def update_train_loop_lr_schedulers(self):
-        if (self.batch_idx + 1) % self.accumulate_grad_batches == 0:
+        if ((self.batch_idx + 1) % self.accumulate_grad_batches == 0 or
+                self.num_training_batches == (self.batch_idx)):
```

Comment:
```suggestion
                self.num_training_batches == (self.batch_idx + 1)):
```
are you sure? num_training_batches is the total count and batch_idx starts from 0. It looks like batch_idx is getting updated at the end of the loop, so I would definitely expect a batch_idx+1 here for the termination criterion. Where does my reasoning go wrong?

Reply:
and can't we use the `is_last_batch` variable? maybe that's anyway safer in case of IterableDataset? What do you think?

Reply:
Yea, you are right. I missed to add 1. already fixed on local.

I will try with `is_last_batch`
Thank you for your suggestion.

Reply:
also one question. Do we need to update lr when we reach the last batch?

Reply:
If a backward pass is happening, I think lr should also update.